DR-2485: Combine JadeTable and LightTable

snf2ye opened the pull request:
## Background
We have two table components in the UI that are almost identical: JadeTable and LightTable. This separation means that we've created features and updated styling in slightly different ways between the two tables. In order to avoid inconsistencies in the UI and to remove the need to update code for tables in multiple places, we should combine these components. 

Changes deployed to my [personal dev environment](https://jade-sh.datarepo-dev.broadinstitute.org/datasets/c5248002-05d6-469e-bf31-38f1792198f3/data):
- Potentially breaking: Any table could have been broken in the process in combining
- Biggest change can be seen in the data view pages for datasets and snapshots

## Changes included in PR
- [x]  LightTableHead converted to Functional Component
- [x] LightTableHead, LightTable, DataView components converted to Typescript
- [x] Removed DatasetTablePreview Component - As far as I can tell it's unused
- [x] Point DataView to use LightTable instead of JadeTable - means that dataset and snapshot data pages now use LightTable. 
- [x] New way of handling case where no rows are displayed in table
- [x] Converted LightTable to use page and rowsPerPage from redux rather than component state
- [x] Move ability to open long values in "see more" dialog box from JadeTable to LightTable
- [x] Move ability to handle null values from JadeTable to LightTable
- [x] Styling Adjustments
- [x] Delete JadeTable and JadeTableHead Component

## Changes in action
With Changes:

https://user-images.githubusercontent.com/13254229/176681536-fc304c4d-587b-4b32-94bd-25366e28ec67.mov

Before Changes:

https://user-images.githubusercontent.com/13254229/176682089-1fe3eb2c-a789-47fb-9f30-08f6dfde36c8.mov

snf2ye committed:
LightTableHead to functional Component

snf2ye committed:
LightTableHead to TS

snf2ye committed:
Convert LightTable to TS

snf2ye committed:
DataView to tsx; move some jadeTable info to lighttable

snf2ye committed:
Basic LightTable working for Data View

snf2ye committed:
use page/rowsPerPage from redux; Set row count (but needs to be updated b/c not quite right - will revisit)

snf2ye committed:
TODO - need to get totalCount

snf2ye committed:
Handle message when no rows

snf2ye committed:
remove DatasetTablePreview - doesn't appear to be used anywhere

snf2ye committed:
Move remaining functionality from JadeTable to LightTable: Handle null and repeated values

snf2ye committed:
Styling changes

snf2ye committed:
Code cleanup

remove unused var

snf2ye committed:
fix snapshot paging bug

add space

snf2ye committed:
Remove JadeTable and JadeTableHead

snf2ye committed:
fix cypress tests

snf2ye committed:
clean up/handle default value for totalRow

nmalfroy commented on src/components/common/data/DataView.tsx:
```diff
@@ -1,46 +1,81 @@
+}: DataViewProps) {
+  const rowKey = (row: TableRowType): string => {
+    const drId = row.datarepo_row_id;
+    return drId ?? row.name;
```
huh...wasn't familiar with this syntax.  What does that do?

snf2ye replied:
> The nullish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is null or undefined, and otherwise returns its left-hand side operand.

For the non-data view tables, datarepo_row_id will be null, so we can just use the row name.

nmalfroy replied:
oh cool.  it's like a coalesce?

snf2ye replied:
I use it as a short-hand for a ternary statement - the above is equivalent to`drId != null ? drId : row.name`.

nmalfroy commented on src/components/common/data/DataView.tsx:
```diff
@@ -81,11 +116,17 @@ function DataView({
+                  columns={columns}
+                  filteredCount={totalRows}
+                  loading={polling}
+                  noRowsMessage="No rows exist in the table" // TODO - handle case where filtered
```
Is this outside the scope of this PR?

snf2ye replied:
Good point - Now should be updated!

nmalfroy reviewed (approved):
Looks awesome!  Code looks good and I'm not sure if you're adding more based on reviews with Lou.  You may want to add a margin at the bottom of the table to makes sure there is clearance on the page

snf2ye committed:
Cleanup styling

- Remove 'summary" variable - doesn't appear to be used
- convert Dataview from grid to flex (just for simplicity)
- Remove Preview Table component (doesn't appear to be used anywhere)
-

lstrano-broad commented:
Styling looks good - met with Shelby on minor tweaks on pagination and table width

okotsopoulos reviewed (approved):
Thanks for taking on this consolidation!  I interacted with your dev environment versus the existing styling, it looks good.  Calling out that with these changes, hovering on a row no longer highlights it, though with the zebra striping it's generally easier to visually connect a row's values across columns.

snf2ye committed:
Get full screen data table on snapshot page (since snapshot page doesn't currently have sidePanel drawer)

snf2ye committed:
Fix up sort: move sort to be right next to col header, make entire col name clickable, make hover work again

snf2ye commented:
@okotsopoulos 
> hovering on a row no longer highlights it

Thank you for catching that!! I added the hover back - it was a property on the JadeTable, but not LightTable.  I didn't realize we had this feature.

snf2ye committed:
handle no rows message when filtering

snf2ye committed:
Add ticket number to TODO comment

snf2ye commented on src/components/common/data/DataView.tsx:
```diff
@@ -0,0 +1,180 @@
+              rowKey={rowKey}
+              rows={rows}
+              searchString={filterStatement}
+              totalCount={totalRows} // TODO - DR-2663 - instead should display total rows regardless of filtering
```
Note: Ticket added to address TODO comment - https://broadworkbench.atlassian.net/browse/DR-2663
